tools/wptrunner/wptrunner/executors/executormarionette.py

```diff
@@ -59,8 +59,16 @@ def setup(self):
         self.marionette = self.parent.marionette
 
     def execute_script(self, script, async=False):
-        method = self.marionette.execute_async_script if async else self.marionette.execute_script
-        return method(script, new_sandbox=False)
+        body = {
+            "script": script.strip(),
+            "args": [],
+            "newSandbox": False,
+            "specialPowers": None,
+            "scriptTimeout": None
+        }
+        method = "WebDriver:ExecuteAsyncScript" if async else "WebDriver:ExecuteScript"
+        rv = self.marionette._send_message(method, body, key="value")
+        return self.marionette._from_json(rv)
```

Comment:
I don't know all this code at all, so maybe I'm not the best reviewer. But I want to state that using the private message `_send_message` from Marionette isn't what I want to see. Can you please explain why this is necessary? 

Also note that a script timeout of None will result in a hang if something goes wrong in a call to an async script.

Reply:
`marionette.py` doesn't expose the possibility to not set a sandbox. This is replicating the message that's set in geckodriver and it's tested in-tree in gecko, so I'm reasonably confident it won't break.

Reply:
@andreastt I would refer to you here. I don't like the `_send_message` usage at all, but maybe you have a better picture for that in regards of wptrunner.

Reply:
Wouldn’t `method(script, sandbox=None, new_sandbox=False)` work?

The problem we’re trying to circumvent here is that the Python client uses the string `default` as the default value for the `sandbox` kwarg, but if this is explicitly set to `None` it will pass `{"sandbox": null, …}` to Marionette.